imgee/models/stored_file.py

```diff
@@ -58,7 +58,29 @@ def __repr__(self):
     def extn(self):
         return guess_extension(self.mimetype, self.orig_extn) or ''
 
-    def is_queued_for_deletion(self):
-        if imgee.app.config.get('CELERY_ALWAYS_EAGER'):
-            return False
-        return imgee.registry.is_queued_for_deletion(self.name+self.extn)
+    def dict_data(self):
+        return dict(
+            title=self.title,
+            uploaded=self.created_at.strftime('%B %d, %Y'),
+            filesize=app.jinja_env.filters['filesizeformat'](self.size),
+            imgsize='%s x %s' % (self.width, self.height),
+            url=url_for('view_image', profile=self.profile.name, image=self.name),
+            thumb_url=url_for('get_image', image=self.name, size=app.config.get('THUMBNAIL_SIZE'))
+        )
+
+    def add_labels(self, labels):
+        new_labels = set(labels)
+        old_labels = set(self.labels)
+        if new_labels != old_labels:
+            self.labels = labels
+
+        if new_labels == old_labels:
+            status, diff = '0', []
+        elif new_labels > old_labels:
+            status, diff = '+', (new_labels - old_labels)
+        elif (old_labels > new_labels):
+            status, diff = '-', (old_labels - new_labels)
+        else:
+            status, diff = '', new_labels
+
+        return status, list(diff)
```

Comment:
This part of the code is a bit hairy. I think it can be handled better. I'll try something and show you.